schema_cache.rst

```diff
@@ -0,0 +1,119 @@
+.. _schema_cache:
+
+Schema Cache
+============
+
+There are several PostgREST features that need information from the database schema.
+If they accessed this information directly from the database every time they needed it, it would be too costly.
+That is why, when PostgREST starts, it generates a database schema cache and uses it to get the information needed for these features:
+
+- For :ref:`resource_embedding`, it looks for foreign key constraints to determine relationships between tables.
+- For :ref:`Database Functions <s_procs>`, it looks for their metadata: parameters, return type, volatility and `overloading <https://www.postgresql.org/docs/current/xfunc-overload.html>`_.
+- For :ref:`Upserts <upsert>`, it looks for the tables' primary keys.
+- For :ref:`insert_update`, it also looks for the tables' primary keys in order to return the Location header.
+- For OPTION requests, it verifies the existence of tables and views and looks for INSTEAD OF triggers for the latter.
+- For :ref:`open-api`, it looks for information about tables (columns, primary keys and foreign keys), views (columns and INSTEAD OF triggers) and functions (metadata).
+
+The Stale Schema Cache
+----------------------
+
+When you make a change in the database related to any of the features mentioned above while PostgREST is running, the schema cache turns stale. You then need to reload the schema before you make a request related to these changes, otherwise you'll receive an error instead of the expected response.
+
+For instance, let's see what would happen if you have a stale schema for foreign key relationships and function metadata:
+
+Stale Foreign Key Relationships
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Suppose you add a ``cities`` table to your database. This table has a foreign key ``country_id`` referencing an existing ``countries`` table. Then, you make a request to get the data from ``cities`` and their ``countries``:
+
+    .. code-block:: http
+
+      GET /cities?select=name,country:countries(id,name) HTTP/1.1
```

Comment:
@laurenceisla This section is showing as indented. I don't think it should, it looks a bit weird. See https://postgrest--404.org.readthedocs.build/en/404/schema_cache.html#stale-foreign-key-relationships.

Reply:
True that, the indentation doesn't look good. It's fixed now.